time/range.go

```diff
@@ -61,6 +61,11 @@ func (r Range) Overlaps(other Range) bool {
 	return r.End.After(other.Start) && r.Start.Before(other.End)
 }
 
+// Duration returns the duration of the range.
+func (r Range) Duration() time.Duration {
+	return r.End.Sub(r.Start)
```

Comment:
Yeah was more worried about the semantic meaning of a range with a negative duration, doesn't matter I guess.